cromshell

```diff
@@ -277,35 +298,67 @@ function execution-status-count() {
 # Bring up a browser window to view timing information on the job.
 function timing() {
   echo "Opeining timing information in a web browser for job ID: ${1}"
-  open ${CROMWELL_URL}/api/workflows/v1/${1}/timing
+  open ${2}/api/workflows/v1/${1}/timing
   return $?
 }
 
 # Cancel a running job.
 function abort() { 
-  response=$(curl --connect-timeout $CURL_CONNECT_TIMEOUT -X POST --header "Content-Type: application/json" --header "Accept: application/json" "${CROMWELL_URL}/api/workflows/v1/${1}/abort")
+  response=$(curl --connect-timeout $CURL_CONNECT_TIMEOUT -X POST --header "Content-Type: application/json" --header "Accept: application/json" "${2}/api/workflows/v1/${1}/abort")
   local r=$?
   echo $response  
   return $r
 }
 
+# List all jobs submitted in ${CROMWELL_SUBMISSIONS_FILE}
+function list() {
+	cat ${CROMWELL_SUBMISSIONS_FILE}
+	return $?
+}
+
+# Get the root log folder from the cloud and put it in the metadata folder for this run
+function fetch() {
+	local id=$1
+	local cromwellServer=$2
+
+	local remoteFolder=$( metadata ${id} ${cromwellServer} | grep "\"callRoot\":" | head -n1 | awk '{print $2}' | sed "s#\"\\(.*${id}\\).*#\\1#g" )
+
+	local localServerFolder="${CROMSHELL_CONFIG_DIR}/$( echo "${cromwellServer}" | sed -e 's#ht.*://##g' )/${id}"
+
+	which gsutil &>/dev/null 	
+	r=$?
+	[ $r -ne 0 ] && error "gsutil does not exist.  Must install google cloud utilities." && exit 8
+
+	mkdir -p ${localServerFolder}
+
+	error "Retrieving logs from ${remoteFolder}"
+	error "Copying into ${localServerFolder}"
+	gsutil rsync -rP ${remoteFolder}/ ${localServerFolder}/.
+
+	error "Files copied to local folder: ${localServerFolder}"
+	
+	return 0
+}
+
 # Spin off a task in the background that will check periodically if the workflow ID is 
 # complete.
 # When it is, send an email to the email address specified.
 function notify() {
 
   local id=$1
   local email=$2
+	local cromwellServer=$3
```

Comment:
accidental tab here

Reply:
Yeah - I need to go through and de-tab the whole thing. 

I have vim to preserve tabs since I read a lot of TSVs.